app/workers/attachments/finish_direct_upload_job.rb

```diff
@@ -29,115 +29,44 @@
 class Attachments::FinishDirectUploadJob < ApplicationJob
   queue_with_priority :high
 
-  def perform(attachment_id, whitelist: true)
+  def perform(attachment_id, allowlist: true)
     attachment = Attachment.pending_direct_upload.find_by(id: attachment_id)
-    # An attachment is guaranteed to have a file.
-    # But if the attachment is nil the expression attachment&.file will be nil and attachment&.file.local_file
-    # will throw a NoMethodError: undefined method local_file' for nil:NilClass`.
-    local_file = attachment && attachment.file.local_file
 
-    if local_file.nil?
-      return Rails.logger.error("File for attachment #{attachment_id} was not uploaded.")
-    end
-
-    User.execute_as(attachment.author) do
-      attach_uploaded_file(attachment, local_file, whitelist)
+    Attachments::FinishDirectUploadService
+      .new(user: attachment.author, model: attachment, contract_options: derive_contract_options(allowlist))
+      .call
+      .on_failure do |call|
+      destroy_attachment_and_log_errors(attachment, call.errors)
```

Comment:
rubocop doesn't seem to care
```suggestion
        destroy_attachment_and_log_errors(attachment, call.errors)
```

Reply:
My editor prevents setting this, so I'll ignore it in this fix